charters/ExpressionMatrixService/charter.md

```diff
@@ -0,0 +1,107 @@
+
+# Expression Matrix Service
+
+## Description
+
+The expression matrix service enables researchers to easily access and analyze HCA expression matrix data.
+
+## Objectives
+
+* Build, operate, and maintain a data access API that serves expression
+matrices from the HCA DCP Data Store with options to subset data based on bundle metadata, genes, and numerical filters.
+
+* Serve as a test bed for a re-usable metadata query service, created by
+the *Query Service* component.
+
+## In-scope
+
+* Expression Matrix Service interface supports queries based on:
+    * Bundle identifiers
+        * This query enables selection of specific submissions.
+    * Metadata (*across bundles; potentially provided directly by Query Module*)
+        * These types of filters will enable users to select cells based on their biological characteristics, for example, all kidney cells, or all cells prepared using the 10x v2 assay.
+    * Numerical filtering (*on gene expression values, and also QC metrics*)
+        * These types of filters will, for example, enable users to query for cells according to their expression of key marker genes, or exclude cells with low quality scores.
+
+* Expression Matrix Service API and scalable backend that provides access to
+query interface
+
+* Architecture, software design, and scale testing for scalability
+
+* Data representation
+    * Multiple expression matrix data formats based on _de facto_ standards delivered by the API and the Command Line Interface (CLI)
+    * Expression matrix data format for storage
+    * QC matrix data format delivered by the API and the Command Line Interface (CLI)
+    * QC matrix data format for storage
+
+
+* Extends HCA DCP Command Line Interface (CLI) client to support the Expression Matrix Service
+
+* Documentation
+    * Publish documentation for API and file formats
+    * Publish example code for developers to enable fast/easy adoption of API by analysis and visualization portals
+
+* User Experience
+    * Conduct UX research and design; drive query and API design directly from use cases
+    * Validate design with communities building portals for analysis and
+    visualization
+    * Provide feedback in the form of architecture sessions and written UX
+    reports to the DCP community about the experience of building layered services using the public API(s). (e.g., *can the expression microservice be built using only public Data Store API?*)
+    * Scientific "guardrails" to warn users when executed queries produce data subsets with questionable scientific utility
+        * For example, if a user queries for the expression of only a single gene across all lung cells, this query would trigger a warning that the returned data cannot be effectively normalized
+
+* External dependencies
```

Comment:
Is there an explicit dependency on the metadata schema team? We call out to gathering requirements from and supporting data consumers to extend the schemas to meet their needs. I would assume you are a specific type data consumer in this context and will be able to provide us with exemplar queries that people want to do.

Reply:
I reviewed with Ambrose and Marcus. We don't expect there to be an explicit dependency. Any related requests on Metadata would be from Secondary Analysis.